contrib/signet/miner

```diff
@@ -406,16 +408,33 @@ def do_generate(args):
 
         # mine block
         logging.debug("Mining block delta=%s start=%s mine=%s", seconds_to_hms(mine_time-bestheader["time"]), mine_time, is_mine)
-        mined_blocks += 1
         psbt = generate_psbt(tmpl, reward_spk, blocktime=mine_time)
         input_stream = os.linesep.join([psbt, "true", "ALL"]).encode('utf8')
         psbt_signed = json.loads(args.bcli("-stdin", "walletprocesspsbt", input=input_stream))
+
         if not psbt_signed.get("complete",False):
             logging.debug("Generated PSBT: %s" % (psbt,))
             sys.stderr.write("PSBT signing failed\n")
             return 1
         block, signet_solution = do_decode_psbt(psbt_signed["psbt"])
-        block = finish_block(block, signet_solution, args.grind_cmd)
+
+        # finish_block() will call bitcoin-util to grind PoW. If the nonce field search space is exhausted,
+        # the subprocess will fail and thus the miner. In case of failure, increment the retries var,
+        # use it to create a new block header and try again.
+        try:
+            block = finish_block(block, signet_solution, args.grind_cmd)
+        except subprocess.CalledProcessError as e:
+            logging.debug('subprocess exception: {}'.format(e))
```

Comment:
I'll mark those as resolved since I simplified the strings in a reviewed commit. Turns out that the script is using this style all over: `logging.debug("Mining block delta=%s start=%s mine=%s", seconds_to_hms(mine_time-bestheader["time"]), mine_time, is_mine)`

Reply:
Ah yes, I feel I should've remembered that 😆 it's usually considered correct-est to use the `%` formatting in log messages because that formatted string is only actually created if the message will be logged based on log-level, vs `f"` or `.format` or `" % ...` which are always created. Others may consider that premature optimisation if string-formatting due to logging has negligible perf impact 😉